src/promptflow-tools/tests/test_open_source_llm.py

```diff
@@ -209,67 +225,53 @@ def test_open_source_llm_llama_parse_chat_with_comp(self):
-    def test_open_source_llm_llama_endpoint_miss(self, gpt2_custom_connection):
-        tmp = copy.deepcopy(gpt2_custom_connection)
-        tmp.configs['endpoint_url'] += 'completely/real/endpoint'
-        os = OpenSourceLLM(tmp)
-        with pytest.raises(OpenSourceLLMOnlineEndpointError) as exc_info:
-            os.call(
-                self.completion_prompt,
-                API.COMPLETION)
-        assert exc_info.value.message == (
-            "Exception hit calling Oneline Endpoint: "
-            + "HTTPError: HTTP Error 424: Failed Dependency")
-        assert exc_info.value.error_codes == "UserError/OpenSourceLLMOnlineEndpointError".split("/")
-
-    @pytest.mark.skip_if_no_api_key("gpt2_custom_connection")
-    def test_open_source_llm_llama_deployment_miss(self, gpt2_provider):
-        with pytest.raises(OpenSourceLLMOnlineEndpointError) as exc_info:
-            gpt2_provider.call(self.completion_prompt,
-                               API.COMPLETION,
-                               deployment_name="completely/real/deployment-007")
-        assert exc_info.value.message == (
-            "Exception hit calling Oneline Endpoint: "
-            + "HTTPError: HTTP Error 404: Not Found")
-        assert exc_info.value.error_codes == "UserError/OpenSourceLLMOnlineEndpointError".split("/")
 
-    @pytest.mark.skip_if_no_api_key("open_source_llm_ws_service_connection")
+    @pytest.mark.skip_if_no_key("open_source_llm_ws_service_connection")
     def test_open_source_llm_chat_endpoint_name(self, chat_endpoints_provider):
         for endpoint_name in chat_endpoints_provider:
-            os_llm = OpenSourceLLM(endpoint_name=endpoint_name)
-            response = os_llm.call(self.chat_prompt, API.CHAT)
+            response = self.stateless_os_llm.call(
+                self.chat_prompt,
+                API.CHAT,
+                connection=f"onlineEndpoint/{endpoint_name}")
             assert len(response) > 25
 
-    @pytest.mark.skip_if_no_api_key("open_source_llm_ws_service_connection")
+    @pytest.mark.skip_if_no_key("open_source_llm_ws_service_connection")
     def test_open_source_llm_chat_endpoint_name_with_deployment(self, chat_endpoints_provider):
         for endpoint_name in chat_endpoints_provider:
-            os_llm = OpenSourceLLM(endpoint_name=endpoint_name)
             for deployment_name in chat_endpoints_provider[endpoint_name]:
-                response = os_llm.call(self.chat_prompt, API.CHAT, deployment_name=deployment_name)
+                response = self.stateless_os_llm.call(
+                    self.chat_prompt,
+                    API.CHAT,
+                    connection=f"onlineEndpoint/{endpoint_name}",
+                    deployment_name=deployment_name)
                 assert len(response) > 25
 
-    @pytest.mark.skip_if_no_api_key("open_source_llm_ws_service_connection")
+    @pytest.mark.skip_if_no_key("open_source_llm_ws_service_connection")
```

Comment:
please kindly use `skip_if_no_api_key` see https://github.com/microsoft/promptflow/pull/935
maybe you should merge/rebase main branch?